### DISCUSSION TITLE
Option to add details in configs

### DISCUSSION THREAD

**Discussion by gk12277** [koxudaxi/datamodel-code-generator#466] [Q&A] [answered]
1.While generation with using JsonSchema is there a option to add different options in [config](https://pydantic-docs.helpmanual.io/usage/model_config/).
2. Also is there a way to specify multiple or difference base class for different classes. I have checked the option to specify base class through cli but it will be common for all the classes. Can we specify  a specific base class or more than one classes for a specific generated class.

**Comment by ag0n1k** [comment]
The reason I'm interesting in 1. is that I need to set `use_enum_values = True`.

**Reply by gk12277** [reply]
@ag0n1k Do you want to set it for all your entities if so u can add it in config in the template. But i don't think there is a option to add config for different entities specifically.

**Reply by ag0n1k** [reply]
@gk12277 
> Do you want to set it for all your entities if so u can add it in config in the template.

Do you have some examples? 

For now I managed this problem by hands and of course I need to set this option only for classes with Enum objects, there is no need for all.

**Reply by koxudaxi** [reply]
@ag0n1k 
You can use a custom template.

This issue may help you.
https://github.com/koxudaxi/datamodel-code-generator/issues/428

original templates are in this directory
https://github.com/koxudaxi/datamodel-code-generator/tree/master/datamodel_code_generator/model/template

**Comment by koxudaxi** [comment, marked as answer]
@gk12277 
I'm sorry for my too-late reply.
You can use extra template data to set config on a specific model.
https://github.com/koxudaxi/datamodel-code-generator/pull/861/files

**Reply by nbro10** [reply]
@koxudaxi So, I'm using 

```
{
    "MyModel": {
        "config": {
            "use_enum_values": "True"
        }
    }
}
```

or even (as e.g. used [here](https://github.com/koxudaxi/datamodel-code-generator/pull/1425))

```
{
    "#all#": {
        "config": {
            "use_enum_values": "True"
        }
    }
}
```

But it's not working, i.e. `use_enum_values` is not being added to the `Config` inner class of `MyModel` (in the first case) or all models in the second. I'm sure the tool is reading the file because if I provide the wrong path to the file I get an error, so this configuration is simply being ignored. So, I suppose this extra config is not support, but wouldn't it be enough just to add it like in this PR: https://github.com/koxudaxi/datamodel-code-generator/pull/1425. So, it seems that this requires a very simple fix and I don't know why this is not supported yet, as this looks like a very important feature.

**Reply by nbro10** [reply]
Ok, I just tried it. Adding `use_enum_values` to `ConfigDict` solves the issue, so my reasoning was correct I think. I don't see why this shouldn't be supported, as we may want to dump our classes with enum values not enum options.
